### ISSUE TITLE
Create queries for internal CCV state

### ISSUE THREAD

**Issue by jtremback** [cosmos/interchain-security#324]
CCV currently does not have many (any?) queries for someone to check what is going on in the internal state of the module. Someone needs to go through the module and decide which internal state it would be convenient to be able to query, then write the boilerplate code to enable it to be queried over CLI and http.

For example, which consumer chains have sent VSCMaturePackets, which consumer chains are even active, etc.

**Comment by MSalopek** [comment]
The list provided below is a starting point for us to discuss which queries we need.

The list is by no means complete and some of the things suggested are not that useful.

To continue on Jehan's instructions above a good start could be:
- listing chains using CCV
- querying unbonding ops (checking maturation timestamps, filtering matured vs pending) 
- listing consumer initiated slashing requests

## Provider
### Available queries
1. consumer-genesis [chain-id]
Dump consumer genesis state.

### Proposed queries

**chains**
- get active consumer chains (pageable)

**clients**
  - get prov height where consumer was initiated
  - check defined unbonding period for chain
  - pending clients
  - spawned clients
  - check client state (client, consensus)


I'm not sure client data should be made available for querying, or if it would be useful if available.

**staking**
  - dump pending unbonding ops/VSC (chain-id, pageable)
  - dump matured unbonding ops (chain-id, pageable)
  - get single unbonding op by ID
  - consumer reported slashing events (chain-id, pageable)

## Consumer
### Available queries
None.

### Proposed

**query validators**
- dump current validator set

**slashing**
- get pending slashing requests (pageable)

**distribution**
- get next dist block height
- pending distribution fee amounts -> total, provider, consumer

Potentially useful for estimating accrued fees to be paid out on next distribution. I'm not sure this is useful to anyone.

**VCS**
- get pending changes (pageable)
- get VSC maturation timestamp (packet-id)

**Comment by mpoke** [comment]
@MSalopek It looks good. Some comments:

- provider / clients / "check defined unbonding period for chain" -- what do you mean by this?
- provider / clients /
> - pending clients
> - spawned clients
> - check client state (client, consensus)
> 
> I'm not sure client data should be made available for querying, or if it would be useful if available.

probably not needed; IBC provides queries for IBC clients

- provider / staking / 
> - dump pending unbonding ops/VSC (chain-id, pageable)
> - dump matured unbonding ops (chain-id, pageable)
> - consumer reported slashing events (chain-id, pageable)

probably not needed

- consumer / slashing / "get pending slashing requests (pageable)" -- probably not needed

**Comment by MSalopek** [comment]
> provider / clients / "check defined unbonding period for chain" -- what do you mean by this?

I thought about being able to learn the the unbonding time duration for consumer chain. On second read, I don't see much use for it
@mpoke

**Comment by mpoke** [comment]
> > provider / clients / "check defined unbonding period for chain" -- what do you mean by this?
> 
> I thought about being able to learn the the unbonding time duration for consumer chain. On second read, I don't see much use for it @mpoke

@jtremback was saying to move the consumer unbonding period to be a param of the CCV module, which means that it should be able to be queried like all other params.

**Comment by jtremback** [comment]
Great job, @MSalopek! I went over this, and looked in the code myself, and this is what I'd like you to start on:

**Provider**
- Active consumer chains
- Pending consumer chain starts
- Pending consumer chain stops

**Consumer**
- get next dist block height
- pending distribution fee amounts -> total, provider, consumer
- dump current validator set

Later:
- @mpoke I think that it should be possible to enable a query on the provider where someone can tell the latest height is that unbonding operations have completed from. So, for example, if I started unbonding at block 10,003, I can see that someone who started unbonding at block 10,002 just had their unbonding complete, so I will be able to get my money out soon. Any thoughts?

I also realized that there are things that people would be interested in that are better suited to events because the state is quite transient and is cleaned up quickly. Stuff like slashings, valset updates, etc etc. But that's in another issue.
